src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -392,11 +392,20 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             vkey != VK_SNAPSHOT &&
             keyDown)
         {
+            if (modifiers.IsCtrlPressed() && vkey == 'A')
+            {
+                auto lock = _terminal->LockForWriting();
+                _terminal->SelectAll();
+                _renderer->TriggerSelection();
+                return true;
+            }
+
             // try to update the selection
-            if (const auto updateSlnParams{ ::Terminal::ConvertKeyEventToUpdateSelectionParams(modifiers, vkey) })
+            if (const auto updateSlnParams{ _terminal->ConvertKeyEventToUpdateSelectionParams(modifiers, vkey) })
             {
                 auto lock = _terminal->LockForWriting();
-                _terminal->UpdateSelection(updateSlnParams->first, updateSlnParams->second);
+                const std::optional<bool> markModeOverride = _terminal->IsInMarkMode() ? modifiers.IsShiftPressed() : std::optional<bool>{};
+                _terminal->UpdateSelection(updateSlnParams->first, updateSlnParams->second, markModeOverride);

```

Comment:
The `UpdateSelection` logic already depends on `_markMode` and now it depends on it twice - once explicitly internally (`_markMode`) and once implicitly (this parameter being dependent on `_markMode`, but computed from "the outside").

You can add a `ControlKeyStates mods` parameter instead and pass `modifiers` directly, the same way you do it with `ConvertKeyEventToUpdateSelectionParams`. Then ignore the parameter depending on the `_markMode` state.

This works well IMO since `ConvertKeyEventToUpdateSelectionParams` already has a very closely related `(_markMode || mods.IsShiftPressed())` condition.